dotnet/src/SemanticKernel.Abstractions/AI/TextCompletion/TextCompletionExtensions.cs

```diff
@@ -39,20 +39,22 @@ public static class TextCompletionExtensions
     /// <param name="text">The prompt to complete.</param>
     /// <param name="requestSettings">Request settings for the completion API</param>
     /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <remarks>This extension does not support multiple prompt results (Only the first will be returned)</remarks>
     /// <returns>Streaming content of the text generated by the remote model</returns>
     public static async IAsyncEnumerable<string> CompleteStreamAsync(this ITextCompletion textCompletion,
         string text,
         CompleteRequestSettings requestSettings,
         [EnumeratorCancellation] CancellationToken cancellationToken = default)
     {
         var completionResults = textCompletion.GetStreamingCompletionsAsync(text, requestSettings, cancellationToken);
-        var firstResult = await completionResults.FirstOrDefaultAsync(cancellationToken).ConfigureAwait(false);
-        if (firstResult is not null)
+
+        await foreach (var completionResult in completionResults)
         {
```

Comment:
Wasn't this more clear before? The new loop doesn't do any looping if it exits at the end of the first iteration

Reply:
When I used the FirstOrDefaultAsync a bug that I cannot understand is happening where it freezes the interation of the message loop. 

Keeping the foreach statement don't surface the bug works as a quick fix as all works as expected.

I can dive more into the problem in another PR, as this might need some more detailed investigation on how the AsyncEnumerable internals work.

Reply:
@stephentoub Thanks for the help, as it was understood using the `FirstOrDefaultAsync` kills the StreamingCompletions object which actually handles the streaming process leaving the inner `foreach` loop in the limbo.

The approach to resolve it would be to either not changing the foreach and keeping the choice yield alive while consuming the inner message/text or making ITextResult IDisposable and handling the disposing down the line.

For the sake of simplicity, I'll keep the outer yield so .net runtime can hold the resource for the time all the inner messages of the first choice are consumed.